joint_trajectory_controller/include/joint_trajectory_controller/init_joint_trajectory.h

```diff
@@ -66,9 +66,6 @@ inline std::vector<unsigned int> permutation(const T& t1, const T& t2)
 {
   typedef unsigned int SizeType;
 
-  // Arguments must have the same size
-  if (t1.size() != t2.size()) {return std::vector<SizeType>();}
```

Comment:
If vectors are no longer to be of the same size, then we are no longer talking about a permutation (rearranging). What's being computed is a map between `t1` indices (implicitly encoded in return vector indices) to `t2` indices. Also, the size of `t1` cannot be larger than that of `t2`. That would be an early optimization.

The function name and documentation should be changed accordingly.